docs/marionette.region.md

```diff
@@ -421,6 +423,28 @@ myRegion.reset();
 
 This can be useful in unit testing your views.
 
+## Check If View Is Being Swapped By Another
+
+The `isSwappingView` method returns if a view is being swapped by another one. It's useful
+inside region lifecycle events / methods.
+
+The example will show an message when the region is empty  
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var EmptyMsgRegion = Mn.Region.extend({
+  onEmpty() {
+    if (!this.isSwappingView()) {
+      this.$el.append('Empty Region');
+    }    
+  }  
+});
+```
+[Live example](https://jsfiddle.net/marionettejs/c1nacq0c/1/)
+
+> `hasView` will not work reliably inside `onEmpty` because will always evaluate to false
```

Comment:
There's no problem with hasView.
The note is to show that for the purpose of checking if a view is about to be rendered hasView is not an option inside onEmpty.

It can be better worded